src/providers/tpm_provider/mod.rs

```diff
@@ -216,6 +217,33 @@ impl TpmProviderBuilder {
         }
     }
 
+    unsafe fn find_default_context_cipher(&self) -> std::io::Result<Cipher> {
```

Comment:
Maybe a comment explaining why `unsafe` is needed and why it is safe under our conditions (I think it is covered by the `Safety` notice of the calling function) would be good.